### PR TITLE
Themes: Switch theme without reload using global shortcut

### PR DESCRIPTION
**Pull request by torkelo**
Adds a new feature to toggle theme using a keyboard shortcut (t t).

https://user-images.githubusercontent.com/10999/111862191-fa538300-8953-11eb-92e7-34d03e288c55.mp4

### CODE REVIEW

File: public/app/core/services/toggleTheme.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  if (runtimeOnly) {
+    return;
+  }
+
+  // Add css file for new theme
+  const newCssLink = document.createElement('link');
+  newCssLink.rel = 'stylesheet';
+  newCssLink.href = config.bootData.themePaths[newTheme.type];
+  document.body.appendChild(newCssLink);
```
Comment (Clarity-89) [inline public/app/core/services/toggleTheme.ts:20]: Should we consider using smth like [react-helmet](https://github.com/nfl/react-helmet) for more declarative document `head` updates?

**Review by Clarity-89 — COMMENTED** [review]
Great job! Noticed that theme switching is inconsistent in panel edits:
Screenshot 2021-03-20 at 12 07 17

Also some parts of the page are very slow to update.

**Comment by torkelo** [comment]
Yea the panel edit issue is likely not updating due to some component using theme directly config and not via provider (context).

That is one reason I wanted to add this is to highlight places like that. There is another shortcut that will only change the emotion theme (not the css file) to highlight all places that are not using emotion

File: public/app/core/services/toggleTheme.ts (hunk 1 above)
Comment (torkelo) [reply public/app/core/services/toggleTheme.ts:20]: Yes, now that the app is fully react we should definitely do that (Especially for updating page title) !!

**Pull request merged**